python-stdlib/argparse/argparse.py

```diff
@@ -10,29 +10,41 @@ class _ArgError(BaseException):
     pass
 
 
+class ArgumentTypeError(BaseException):
+    pass
+
+
 class _Arg:
-    def __init__(self, names, dest, action, nargs, const, default, help):
+    def __init__(self, names, dest, action, nargs, const, default, help, type):
         self.names = names
         self.dest = dest
         self.action = action
         self.nargs = nargs
         self.const = const
         self.default = default
         self.help = help
+        self.type = type
+
+    def _apply(self, optname, arg):
+        if self.type:
+            try:
+                return self.type(arg)
+            except Exception as e:
+                if isinstance(e, (ArgumentTypeError, TypeError, ValueError)):
+                    raise _ArgError("invalid value for %s: %s (%s)" % (optname, arg, str(e)))
+                raise
```

Comment:
How about writing this as more idiomatic Python:
```py
except (ArgumentTypeError, TypeError, ValueError) as e:
    raise _ArgError(...)
```
I think that's equivalent?  It also saves 12 bytes in the resulting .mpy.

Reply:
I thought this syntax wasn't supported by MicroPython, good to know I was wrong all along :)

It is indeed equivalent.  Thanks for pointing this out, I've changed the code to reflect that.